### PR TITLE
Hw04 auth

### PR DESCRIPTION
**Pull request by KhailoAndrey**
add: sort, pagination, current user & some fix

### CODE REVIEW

File: middlewares/userMiddlewares.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  if (!user) return next(new AppError(404, 'User does not exist..'));
+
+  req.user = user;
+
+  next();
+});
+
+const checkCreateUserData = catchAsync(async (req, res, next) => {
+  console.log('req.body --->', req.body)
```
Comment (ceREAL133) [inline middlewares/userMiddlewares.js]: delete log)

File: middlewares/authMiddlewares.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  } catch (err) {
+    console.log(err.message);
+
+    throw new AppError(401, 'Not logged in!');
+  }
+
+  const currentUser = await User.findById(decoded.id);
+
+  if (!currentUser) throw new AppError(401, 'Not logged in!');
+console.log('middleware-->', currentUser)
```
Comment (ceREAL133) [inline middlewares/authMiddlewares.js]: log

File: controllers/authController.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    user.token = token;
+
+    res.status(200).json({
+        user,
+    })
+})
+
+exports.logout = catchAsync(async (req, res) => { 
+    console.log(req.user)
```
Comment (ceREAL133) [inline controllers/authController.js]: log

**Review by ceREAL133 — APPROVED** [review]

File: middlewares/userMiddlewares.js (hunk 1 above)
Comment (KhailoAndrey) [reply middlewares/userMiddlewares.js]: done

File: middlewares/authMiddlewares.js (hunk 2 above)
Comment (KhailoAndrey) [reply middlewares/authMiddlewares.js]: done

File: controllers/authController.js (hunk 3 above)
Comment (KhailoAndrey) [reply controllers/authController.js]: done

**Pull request merged**